File: typescript/src/patterns/factory/ConcreteProducts.ts
import type { Product } from './Product.js';

export class BasicProduct implements Product {
  name() { return 'BasicProduct'; }
  compute(price: number) { return price; }
}

export class PremiumProduct implements Product {
  name() { return 'PremiumProduct'; }
  compute(price: number) { return price * 1.25; }
}

File: python/src/patterns/factory/concrete_products.py
from .product import Product

class BasicProduct(Product):
    def name(self) -> str:
        return "BasicProduct"
    def compute(self, price: float) -> float:
        return float(price)

class PremiumProduct(Product):
    def name(self) -> str:
        return "PremiumProduct"
    def compute(self, price: float) -> float:
        return float(price) * 1.25
